js/src/dapps/registry/Reverse/reverse.js

```diff
@@ -0,0 +1,131 @@
+// Copyright 2015, 2016 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { Card, CardHeader, CardText } from 'material-ui/Card';
+import TextField from 'material-ui/TextField';
+import DropDownMenu from 'material-ui/DropDownMenu';
+import MenuItem from 'material-ui/MenuItem';
+import RaisedButton from 'material-ui/RaisedButton';
+import AddIcon from 'material-ui/svg-icons/content/add';
+import CheckIcon from 'material-ui/svg-icons/navigation/check';
+
+import { propose, confirm } from './actions';
+import styles from './reverse.css';
+
+class Reverse extends Component {
+
+  static propTypes = {
+    pending: PropTypes.bool.isRequired,
+    queue: PropTypes.array.isRequired,
+
+    propose: PropTypes.func.isRequired,
+    confirm: PropTypes.func.isRequired
+  }
+
+  state = {
+    action: 'propose',
+    name: '',
+    address: ''
+  };
+
+  render () {
+    const { pending } = this.props;
+    const { action, address, name } = this.state;
+
+    return (
+      <Card className={ styles.reverse }>
+        <CardHeader title={ 'Manage Reverse Names' } />
+        <CardText>
+          <p className={ styles.noSpacing }>
+            <strong>
+              To make others to find the name of an address using the registry, you can propose & confirm reverse entries.
+            </strong>
+          </p>
+          { action === 'propose' ? (
+            <p className={ styles.noSpacing }>
+              To propose a reverse entry for <code>foo</code>, you have to be the owner of it.
+            </p>
+          ) : (
+            <p className={ styles.noSpacing }>
+              To confirm a proposal, send the transaction from the account that the name has been proposed for.
+            </p>
+          ) }
+          <div className={ styles.box }>
+            <DropDownMenu
+              disabled={ pending }
+              value={ action }
+              onChange={ this.onActionChange }
+            >
+              <MenuItem value='propose' primaryText='propose a reverse entry' />
+              <MenuItem value='confirm' primaryText='confirm a reverse entry' />
+            </DropDownMenu>
+            { action === 'propose' ? (
```

Comment:
As per above, slightly more readable since there is a null for one condition, but the reader and maintainer gets lost here)